src/puppetlabs/pcp/message.clj

```diff
@@ -198,7 +198,7 @@
           data-frame (or (:data data-chunk) (byte-array 0))
           data-flags (or (get-in data-chunk [:descriptor :flags]) #{})]
       (try+ (s/validate Envelope envelope)
-            (catch Throwable _
+            (catch Object _
```

Comment:
It was a bug actually. Without it, this test fails: https://github.com/puppetlabs/clj-pcp-common/blob/ed5727b18c5033abdd130b17a27761003461cd05/test/puppetlabs/pcp/message_test.clj#L117-L120
because the validation exception is propagated directly as it's not caught by the `(catch Throwable _)`